trie/database.go

```diff
@@ -293,34 +152,37 @@ func NewDatabaseWithConfig(diskdb ethdb.Database, config *Config) *Database {
 	if config != nil && config.Preimages {
 		preimage = newPreimageStore(diskdb)
 	}
-	db := &Database{
-		diskdb: diskdb,
-		cleans: cleans,
-		dirties: map[common.Hash]*cachedNode{{}: {
-			children: make(map[common.Hash]uint16),
-		}},
+	var resolver ChildResolver
```

Comment:
Any particular reason for ChildResolver and MPTResolver to be public?

Reply:
Because I plan to move the trie.Database to a separate package(maybe `trie/triedb`) and put all pbss stuff there(maybe `trie/triedb/pbss`) as well. So I make this resolve publicly.